pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -827,32 +827,31 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 	// If we need to (re-)create the pod sandbox, everything will need to be
 	// killed and recreated, and init containers should be purged.
 	if createPodSandbox {
-		if !shouldRestartOnFailure(pod) && attempt != 0 && len(podStatus.ContainerStatuses) != 0 {
```

Comment:
I meant here:
if the pod restartPolicy is Never but the regular containers are still running, we might need to restart sidecar container, at some point.

Reply:
this PR made a fix to cover that case.

Reply:
I'll just fix the sidecar branch and close this PR so that a pod doesn't recreate the pod sandbox if there is no regular container to start ignoring the sidecar containers.

ref: https://github.com/SergeyKanzhelev/kubernetes/pull/21/files#r1212452995